yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -94,6 +98,14 @@ private[yarn] class YarnAllocator(
   // Additional memory overhead.
   protected val memoryOverhead: Int = sparkConf.getInt("spark.yarn.executor.memoryOverhead",
     math.max((MEMORY_OVERHEAD_FACTOR * executorMemory).toInt, MEMORY_OVERHEAD_MIN))
+
+  // Make the maximum executor failure check to be relative with respect to duration
+  private val relativeMaxExecutorFailureCheck = 
```

Comment:
@srowen if you're suggesting tying it to the dstream window I think that could be confusing.  It doesn't seem obvious to me that these should be proportional.

Another thing is that it seems confusing that, if I want to set a max failures per time interval, I need to set three different properties.  It's also worth considering just adding a `spark.yarn.max.executor.failuresPerMinute` property.